### DISCUSSION TITLE
Inconsistent Direction Names in Split Commands

### DISCUSSION THREAD

**Discussion by adiepenbrock** [ghostty-org/ghostty#2986] [Feature Requests, Ideas] [answered]
## Description
The keybind configuration for split-related commands currently uses inconsistent terminology for vertical directions:

- `new_split` uses: `up`, `down`, `left`, `right`
- `goto_split` uses: `top`, `bottom`, `left`, `right`

The horizontal directions (left/right) are consistent between both commands, but vertical directions use different terms (up/down vs top/bottom).

## Expected Behavior
The commands should use consistent terminology for directions. We should either:

- Standardize on up/down for both commands
- Standardize on top/bottom for both commands

## Questions

- Is there any intentional reason for this naming difference?
- Which terminology set would be more intuitive for users?

## Possible (additional) Extension
While addressing the naming inconsistency, we could extend the functionality of `goto_split` to support both absolute and relative vertical navigation:

- Keep `top` and `bottom` for absolute navigation (jump to the topmost/bottommost pane)
- Add `up` and `down` for relative navigation (move one pane up/down from current position)

**Comment by Xophmeister** [comment]
To add to this, `goto_split:{top,bottom,left,right}`, regardless of the direction names, doesn't appear to consistently navigate in a way that matches the layout of the splits. It seems to be dependent on the layout, but one can easily construct a layout where `top`/`left` both cycle to the previous split, whereas `bottom`/`right` cycle to the next.

(Note: Also discussed here: https://github.com/ghostty-org/ghostty/discussions/3604)

**Comment by mitchellh** [comment, marked as answer]
Fixed on main
